src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -1054,6 +1065,40 @@ namespace winrt::TerminalApp::implementation
         _TaskbarProgressChangedHandlers(nullptr, nullptr);
     }
 
+    // Method Description:
+    // - Set an indicator on the tab if any pane is in a closed connection state.
+    // - Show/hide the Restart Connection context menu entry depending on active pane's state.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TerminalTab::_UpdateConnectionClosedState()
+    {
+        ASSERT_UI_THREAD();
+
+        if (_rootPane)
+        {
+            const bool isClosed = _rootPane->WalkTree([&](const auto& p) {
+                return p->IsConnectionClosed();
+            });
+
+            _tabStatus.IsConnectionClosed(isClosed);
+        }
+
+        if (_activePane)
+        {
+            _restartConnectionMenuItem.Visibility(_activePane->IsConnectionClosed() ?
+                                                      WUX::Visibility::Visible :
+                                                      WUX::Visibility::Collapsed);
+        }

```

Comment:
The visibility of the item depends on the current `_activePane`. So does this code need to run when `_UpdateActivePane` is called?

Reply:
Indeed `_UpdateActivePane` calls this `_UpdateConnectionClosedState` function